src/context.rs

```diff
@@ -2598,4 +2623,67 @@ mod tests {
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_drop_function() -> Result<()> {
+        let sf_context = in_memory_context().await;
+
+        let err = sf_context
+            .plan_query(r#"DROP FUNCTION nonexistentfunction"#)
+            .await
+            .unwrap_err();
+
+        assert_eq!(
+            err.to_string(),
+            "Error during planning: Function \"nonexistentfunction\" not found"
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_drop_function_if_exists() -> Result<()> {
+        let sf_context = in_memory_context().await;
+
+        // Source: https://gist.github.com/going-digital/02e46c44d89237c07bc99cd440ebfa43
+        let plan = sf_context
+            .plan_query(r#"DROP FUNCTION IF EXISTS nonexistentfunction"#)
+            .await;
+        assert!(plan.is_ok());
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_create_and_drop_two_functions() -> Result<()> {
+        let sf_context = in_memory_context().await;
+
+        let create_function_stmt = r#"CREATE FUNCTION sintau AS '
+        {
+            "entrypoint": "sintau",
+            "language": "wasm",
+            "input_types": ["int"],
+            "return_type": "int",
+            "data": "AGFzbQEAAAABDQJgAX0BfWADfX9/AX0DBQQAAAABBQQBAUREBxgDBnNpbnRhdQAABGV4cDIAAQRsb2cyAAIKjgEEKQECfUMAAAA/IgIgACAAjpMiACACk4siAZMgAZZBAEEYEAMgAiAAk5gLGQAgACAAjiIAk0EYQSwQA7wgAKhBF3RqvgslAQF/IAC8IgFBF3ZB/wBrsiABQQl0s0MAAIBPlUEsQcQAEAOSCyIBAX0DQCADIACUIAEqAgCSIQMgAUEEaiIBIAJrDQALIAMLC0oBAEEAC0Q/x2FC2eATQUuqKsJzsqY9QAHJQH6V0DZv+V88kPJTPSJndz6sZjE/HQCAP/clMD0D/T++F6bRPkzcNL/Tgrg//IiKNwBqBG5hbWUBHwQABnNpbnRhdQEEZXhwMgIEbG9nMgMIZXZhbHBvbHkCNwQAAwABeAECeDECBGhhbGYBAQABeAICAAF4AQJ4aQMEAAF4AQVzdGFydAIDZW5kAwZyZXN1bHQDCQEDAQAEbG9vcA=="
+        }';"#;
+
+        let create_function_stmt2 = r#"CREATE FUNCTION sintau2 AS '
+        {
+            "entrypoint": "sintau",
+            "language": "wasm",
+            "input_types": ["int"],
+            "return_type": "int",
+            "data": "AGFzbQEAAAABDQJgAX0BfWADfX9/AX0DBQQAAAABBQQBAUREBxgDBnNpbnRhdQAABGV4cDIAAQRsb2cyAAIKjgEEKQECfUMAAAA/IgIgACAAjpMiACACk4siAZMgAZZBAEEYEAMgAiAAk5gLGQAgACAAjiIAk0EYQSwQA7wgAKhBF3RqvgslAQF/IAC8IgFBF3ZB/wBrsiABQQl0s0MAAIBPlUEsQcQAEAOSCyIBAX0DQCADIACUIAEqAgCSIQMgAUEEaiIBIAJrDQALIAMLC0oBAEEAC0Q/x2FC2eATQUuqKsJzsqY9QAHJQH6V0DZv+V88kPJTPSJndz6sZjE/HQCAP/clMD0D/T++F6bRPkzcNL/Tgrg//IiKNwBqBG5hbWUBHwQABnNpbnRhdQEEZXhwMgIEbG9nMgMIZXZhbHBvbHkCNwQAAwABeAECeDECBGhhbGYBAQABeAICAAF4AQJ4aQMEAAF4AQVzdGFydAIDZW5kAwZyZXN1bHQDCQEDAQAEbG9vcA=="
+        }';"#;
+
+        // Create two functions in two separate passes
+        sf_context.plan_query(create_function_stmt).await?;
+        sf_context.plan_query(create_function_stmt2).await?;
+
+        // Test dropping both functions in one pass
+        let plan = sf_context
+            .plan_query(r#"DROP FUNCTION sintau, sintau2"#)
```

Comment:
What happens if you do `DROP FUNCTION sintau, sintau`? 🙃 (I think it just runs a SQL `DELETE ... WHERE function_name IN ('sintau', 'sintau')` and it'll be fine)

Reply:
Good shout. Just applied that edit
![image](https://github.com/splitgraph/seafowl/assets/182515/0275d5f0-2c2d-4371-94ee-aa5d8a44fe5a)
and invoked the test
![image](https://github.com/splitgraph/seafowl/assets/182515/30ac656b-f47a-42a4-8ff2-7a8e47bf24eb)
and it still passes. 

So yeah, I think you're right it's fine.